src/serial.rs

```diff
@@ -840,6 +873,22 @@ mod tests {
         assert_eq!(serial.events.read_count.count(), 1);
         assert_eq!(serial.events.out_byte_count.count(), 1);
         assert_eq!(serial.events.tx_lost_byte_count.count(), 1);
+
+        // This DATA read should cause the `SerialEvents::buffer_read` method to be invoked.
+        // And since it was the last byte from in buffer the `SerialEvents::in_buffer_empty`
```

Comment:
Can we also test that `in_buffer_empty` is not called if there are more bytes in the buffer?